### DISCUSSION TITLE
Slow when logged in the API SERVER

### DISCUSSION THREAD

**Discussion by VenimK** [infiniteremote/installer#4] [Q&A]
When I logged in the API server thing are really slow of time-outs....
No problem with connections when logged out....

**Comment by inkcmd** [comment]
It looks like we won't get an answer here

**Reply by VenimK** [reply]
There .s a fix for by editing a source file
I.l try to check for you
And keep you updated

**Reply by inkcmd** [reply]
initially i thought that the root of this problem was in win10 or 64 bit client, or DNS, i tried to turn off the "key" parameter of hbbs, but the key is mandatory, as i could understand from official site, and maybe connections became unsafe, it doesn't fix connection time with logged in API.
hopefully waiting for your reply. thank you.

**Reply by VenimK** [reply]
in the src/client.rs file

search for these 

if !key.is_empty() && !token.is_empty() {

you 'll fnd that on line 322
and on line 638

this is the code snipet you gotta edit

if !key.is_empty() && !token.is_empty() {
                // mainly for the security of token
                allow_err!(secure_tcp(&mut socket, key).await);
            }
            
            
 You have to comment those out
 
 some it becomes this
  /*    
            if !key.is_empty() && !token.is_empty() {
                // mainly for the security of token
                allow_err!(secure_tcp(&mut socket, key).await);
            }
            */
            
            do that for both lines
            and build your  Rustdesk client again

**Reply by gitondezhuts** [reply]
thanks @VenimK

**Reply by inkcmd** [reply]
Yes, its working with android and windows. But for macos this timeouts are key elements I think, cuz app doesn't want to work there. But we got not so much machines with macos, so it doesn't matter. Thanks

**Reply by VenimK** [reply]
do you build it on osx to then ??

**Reply by inkcmd** [reply]
Build it on github actions. Works with timeout lines, but not with this lines. No error, just don't want to start

**Reply by VenimK** [reply]
there must be some errors on it ....
must be a sciter thingie then
